judge/models/problem.py

```diff
@@ -451,6 +451,32 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def solved_by(self, user):
+        # Return true if a full AC submission to the problem from the user exists.
+        return self.submission_set.filter(user=user.profile, result='AC', points=F('problem__points')).exists()
+
+    # 0 = nothing, 1 = view only, 2 = edit and view
+    def get_vote_perm(self, user):
+        if not user.is_authenticated:
+            return 0
```

Comment:
Let's consider this more thoroughly. Why should anonymous users not be able to see join stats. Might seem weird, but I'd liken it to anonymous users not being able to see editorials.

Reply:
Out of band: removing for anon users reduces clutter, so let's keep it this way.